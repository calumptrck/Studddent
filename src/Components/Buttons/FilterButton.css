
.filterButton {
  float: left;
  padding-right: 30px;
  font-size: 15px;
  font-weight: bold;
  color: rgb(135, 135, 135);
  letter-spacing: 0.6px;
  cursor: pointer;
  transition: all .26s cubic-bezier(0,.5,.51,1);
  background: none;
  border: none;
  border-radius: 1.5em;
  padding: 10px 18px 10px 18px;
  letter-spacing: 0.9px;
  margin-left: 10px;
}

.active {
  background-image: linear-gradient(-180deg,#FF6A57 0,#FF2F68 100%);
  color: #fff;
  box-shadow: 0 15px 25px rgba(0,0,0,0.06);
}

.filterButton:hover {
  background-image: linear-gradient(-180deg,#FF6A57 0,#FF2F68 100%);
  color: #fff;
  box-shadow: 0 15px 25px rgba(0,0,0,0.06);
  transform: scale(1.06);
}

button:focus {outline:0;}

@media screen and (max-width: 992px) {
  .filters {
    margin-left: 15px;
  }

  .filterButton {
    margin-left: 0px;
    margin-right: 15px;
  }

  .filters > :last-child {margin-right: 0 !important; }
}

@media screen and (max-width: 537px) {
  .filters {
    margin-left: 0px;
    
  }
  .filterButton {
    margin-right: 0px;
    font-size: 12px;
    padding: 10px 14px 10px 14px;
  }
}