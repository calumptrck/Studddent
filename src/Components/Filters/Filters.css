.filtersContainer {
  margin-top: 40px;
  display: block;
  height: 120px;
}

.filters {
  margin-top: 10px;
  float: right;
}


input {
  float: left;
}

@media screen and (max-width: 992px) {
  .filters {
    float: none;
    margin-left: 5px;
  }
}
