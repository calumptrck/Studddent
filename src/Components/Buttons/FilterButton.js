import React from 'react';
import './FilterButton.css'

const FilterButton = ({children, id, className, onClick}) =>
    <button type="button" className={className} id={id} onClick={onClick}>{children}</button>

export default FilterButton