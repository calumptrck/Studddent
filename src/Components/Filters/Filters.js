import React from 'react';
import './Filters.css'

import Search from './Search/Search';
import FilterButton from '../Buttons/FilterButton';

const Filters = ({buttons, buttonClick, searchUpdate}) =>
  <div className="container filtersContainer">
    <Search searchUpdate={searchUpdate} />
    <div className="filters">
    {buttons.map((button) =>
      <FilterButton 
        className={"filterButton" + (button.active ? " active" : " default")}
        key={button.id}
        onClick={() => buttonClick(button.id)}
        >{button.label}
        </FilterButton>
    )}
    </div>
  </div>


export default Filters;