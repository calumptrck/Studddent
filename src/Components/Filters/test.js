import React from 'react';
import ReactDOM from 'react-dom';
import Filters from './Filters';
import renderer from 'react-test-renderer';
import FilterButton from '../Buttons/FilterButton';



const buttons = [
  {
    id: 0,
    label: "ALL",
    active: true,
  },
  {
    id: 1,
    label: "DESIGN",
    active: false,
  },
  {
    id: 2,
    label: "DEVELOPMENT",
    active: false,
  },
  {
    id: 3,
    label: "UTILITY",
    active: false,
  }
]

const buttonClick = function(id) {
  for (let i = 0; i < buttons.length; i++) {
    if (i === id) {
      buttons[i].active = true;
    } else {
      buttons[i].active = false;
    }
  }
}

describe('Filters', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Filters buttons={buttons} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('Has one active button', () => {
    const expected = [{
      id: 2,
      label: "DEVELOPMENT",
      active: true,
    },
    {
      id: 0,
      label: "ALL",
      active: false,
    }]
    const component = renderer.create(
      <FilterButton onClick={buttonClick(2)}/>);
    expect(buttons).toEqual(expect.arrayContaining(expected));
  });
});