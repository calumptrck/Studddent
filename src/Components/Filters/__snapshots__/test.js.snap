// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters Has one active button 1`] = `
<button
  className={undefined}
  id={undefined}
  onClick={undefined}
  type="button"
/>
`;
